### ISSUE TITLE
String representing curly braces is incorrectly detected

### ISSUE THREAD

**Issue by BrianSidebotham** [warpnet/salt-lint#192]
**Describe the bug**

In a string with curly brace-notation for standard string formatting it is necessary to use double curly braces in order to output a pair of single curly braces.

The following snippet is an extract from a jinja file to format an ldap search pattern.

```jinja
{% set search_filter="(&({}={{0}})({}=CN={},{}))".format(string1, string2, string3, string4)) %}
```

The output should be (assuming the string contents are the same as the variable name) like so:

```python
search_filter="(&(string1={0})(string2=CN=string3,string4))"
```
However, salt-lint picks up the double braces as a jinja replacement. In this case, jinja will not replace, it formats the string as expected.

Quite a conundrum

**To Reproduce**

Use the string identified above in a `jinja` file and then run salt-lint.

**Expected behavior**

The string format to pass lint.

**Desktop (please complete the following information):**

any

**Comment by roaldnefs** [comment]
Thanks @BrianSidebotham for you bug report! I'll try to fix this issue in the upcoming days.

**Comment by BrianSidebotham** [comment]
@roaldnefs No problem.

This one is a little tricky to get right. I started with a quick bodge in `JinjaVariablesHasSpacesRule.py`:

```python
    def match(self, file, line):
        return self.bracket_regex.search(re.sub("{{\d+}}", "", line))
```

Marginally better, but nowhere near complete. Technically this double quoted sequence can only exist inside a string object (single, double or triple quoted).

I can work on a PR if you like though.

Oh, there's also an error in the test suite because `setUp()` is called before every test and that results in adding the same rule to the list over and over again. I had to modify the linter `RulesCollection.register()` function to stop that happening:

```python
    def register(self, obj):
        if not any(rule.id == obj.id for rule in self.rules):
            self.rules.append(obj)
```

Let me know if you want me to PR or if you're happy to continue.

**Comment by roaldnefs** [comment]
@BrianSidebotham, feel free to take a stab at making a pull request! Please let me know if you need any help.
